bitrepository-core/src/main/java/org/bitrepository/common/utils/SettingsUtils.java

```diff
@@ -101,41 +101,51 @@ public static String getCollectionName(String collectionID) {
         return name;
     }
 
+    public static IntegrityServiceSettings getIntegrityServiceSettings() {
+        return settings.getReferenceSettings().getIntegrityServiceSettings();
+    }
+
+    /**
+     * Grab all PillarDetails from the ReferenceSettings if any are defined and return them as a list.
+     * Return an empty list if not defined.
+     * @return List of all PillarDetails defined in the ReferenceSettings
+     */
+    public static List<PillarIntegrityDetails.PillarDetails> getPillarIntegrityDetails() {
+        PillarIntegrityDetails pillarIntegrityDetails = getIntegrityServiceSettings().getPillarIntegrityDetails();
+        if (pillarIntegrityDetails != null) {
+            List<PillarIntegrityDetails.PillarDetails> pillarDetails = pillarIntegrityDetails.getPillarDetails();
+            if (pillarDetails != null) {
+                return pillarDetails;
+            }
+        }
+        return new ArrayList<>();
```

Comment:
Since we are not promising a mutable list (which I also would not see as helpful), I would prefer `return List.of();` (which returns an immutable list).